src/ostree/ot-builtin-commit.c

```diff
@@ -582,6 +651,10 @@ ostree_builtin_commit (int argc, char **argv, GCancellable *cancellable, GError
     {
       gboolean update_summary;
       guint64 timestamp;
+      g_autoptr(GVariant) old_metadata = g_steal_pointer (&metadata);
+
+      fill_bindings (repo, old_metadata, &metadata);
```

Comment:
I’ve made these changes locally (Krzesimir’s contract with Endless has expired for now).

> The handling here is special enough (and we have enough non-commandline commit writing programs) though that I'm wondering if we should actually lower this logic into `ostree_repo_write_commit_with_opts()` or something and have one of those options be "refs"?

Maybe. I would prefer to do that as a follow-up commit though, rather than introducing more new API here.